### ISSUE TITLE
Implement @Destructor annotation

### ISSUE THREAD

**Issue by hexaredecimal** [hexaredecimal/BlazingWebX#5]
This will oppose the `@Initializer` annotation in that it will run when the program is stopped.

**Comment by hexaredecimal** [comment]
Possible fix: https://adambien.blog/roller/abien/entry/sigterm_sigint_sigkill_and_java
```java
public class App {

    public static void main(String[] args) {

        var shutdownListener = new Thread(){
            public void run(){
                System.out.println("shutdown in 10s ");
                try {
                    Thread.sleep(10000);
                } catch (InterruptedException e) {}
            }
        };
        Runtime.getRuntime().addShutdownHook(shutdownListener);

        Executors.newSingleThreadScheduledExecutor().
                scheduleAtFixedRate(() -> System.out.println("."), 0, 2, TimeUnit.SECONDS);
        
    }
}   
````

**Comment by hexaredecimal** [comment]
This will work for SIGTERM, SIGINT (e.g. CTRL+C) signals, but with SIGKILL the JVM is killed immediately.

The [SIGTERM](https://en.wikipedia.org/wiki/Signal_(IPC)) signal requests termination, whereby [SIGKILL](https://en.wikipedia.org/wiki/Signal_(IPC)) "...is sent to a process to cause it to terminate immediately..."

**Comment by hexaredecimal** [comment]
Implemented in dc9c3d4a5373b0380651ccd2a13c72e9e47865a4
There are still some issues with the implementation but I am closing this issue. 
Improvements will now be made to the base implementation.
